server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -269,7 +269,7 @@ public NumberFieldMapper build(MapperBuilderContext context) {
                 dimension.setValue(true);
             }
 
-            MappedFieldType ft = new NumberFieldType(context.buildFullName(leafName()), this);
+            MappedFieldType ft = new NumberFieldType(context.buildFullName(leafName()), this, context.isSourceSynthetic());
```

Comment:
I have to do this because this PR will be backported and in 8.x `_source.mode` parameter is still operational.